server/neptune/workflows/internal/deploy/terraform/runner.go

```diff
@@ -50,15 +57,14 @@ func (r *WorkflowRunner) Run(ctx workflow.Context, deploymentInfo DeploymentInfo
 }
 
 func (r *WorkflowRunner) awaitWorkflow(ctx workflow.Context, future workflow.ChildWorkflowFuture, deploymentInfo DeploymentInfo) error {
-	var childWE workflow.Execution
-	if err := future.GetChildWorkflowExecution().Get(ctx, &childWE); err != nil {
-		return errors.Wrap(err, "getting child workflow execution")
-	}
-
-	selector := workflow.NewSelector(ctx)
+	selector := workflow.NewNamedSelector(ctx, "TerraformChildWorkflow")
 
-	// our child workflow will signal us when there is a state change which we will
-	// handle accordingly
+	// our child workflow will signal us when there is a state change which we will handle accordingly.
+	// if for some reason the workflow is orphaned or we are retrying it independently, we have no way
```

Comment:
Can you explain an example of how the child wf gets orphaned? I imagine we would want to fail any running terraform workflows if the parent deploy fails.

Just wanted to highlight that there is a field in the ChildWorkflowOptions that also gives us the ability to cancel a workflow if the parent workflow fails for some reason: https://github.com/temporalio/api-go/blob/v1.12.0/enums/v1/workflow.pb.go#L87

Reply:
let's say we need to kill it manually, and then we restart using the commandline, it'd be orphaned and have no where to report its state.